yarn-project/txe/src/oracle/txe_oracle.ts

```diff
@@ -656,38 +658,28 @@ export class TXE implements TypedOracle {
   }
 
   private async executePublicFunction(args: Fr[], callContext: CallContext) {
-    const execution = new PublicExecutionRequest(callContext, args);
+    const executionRequest = new PublicExecutionRequest(callContext, args);
 
     const db = await this.trees.getLatest();
-    const previousBlockState = await this.#getTreesAt(this.blockNumber - 1);
-
-    const combinedConstantData = CombinedConstantData.empty();
-    combinedConstantData.globalVariables.chainId = this.chainId;
-    combinedConstantData.globalVariables.version = this.version;
-    combinedConstantData.globalVariables.blockNumber = new Fr(this.blockNumber);
-    combinedConstantData.historicalHeader.globalVariables.chainId = this.chainId;
-    combinedConstantData.historicalHeader.globalVariables.version = this.version;
-    combinedConstantData.historicalHeader.globalVariables.blockNumber = new Fr(this.blockNumber - 1);
-    combinedConstantData.historicalHeader.state = await db.getStateReference();
-    combinedConstantData.historicalHeader.lastArchive.root = Fr.fromBuffer(
-      (await previousBlockState.getTreeInfo(MerkleTreeId.ARCHIVE)).root,
-    );
 
-    combinedConstantData.txContext.chainId = this.chainId;
-    combinedConstantData.txContext.version = this.version;
+    const globalVariables = GlobalVariables.empty();
+    globalVariables.chainId = this.chainId;
+    globalVariables.chainId = this.chainId;
+    globalVariables.version = this.version;
+    globalVariables.blockNumber = new Fr(this.blockNumber);
 
-    const simulator = new PublicExecutor(
+    const simulator = new PublicTxSimulator(
+      db,
       new TXEWorldStateDB(db, new TXEPublicContractDataSource(this)),
       new NoopTelemetryClient(),
+      globalVariables,
+      /*realAvmProvingRequests=*/ false,
```

Comment:
If at some point we could avoid this flag and maybe use a non-proving subclass here or something like that, it might be better.